xcodeproj/internal/bazel_integration_files/calculate_output_groups.py

```diff
@@ -96,49 +131,95 @@ def _calculate_guid_target_ids(base_objroot):
-            value
-            for configuration in target_pif["buildConfigurations"]
-            for key, value in configuration["buildSettings"].items()
-            if key.startswith("BAZEL_TARGET_ID")
-        ]
+        build_target_ids = {"key": "BAZEL_TARGET_ID"}
+        compile_target_ids = {"key": "BAZEL_COMPILE_TARGET_ID"}
+        for configuration in target_pif["buildConfigurations"]:
+            for key, value in configuration["buildSettings"].items():
+                if key.startswith("BAZEL_TARGET_ID"):
+                    build_target_ids[_platform_from_build_key(key)] = value
+                elif key.startswith("BAZEL_COMPILE_TARGET_ID"):
+                    compile_target_ids[_platform_from_compile_key(key)] = value
+
+        target_ids = {
+            "build": build_target_ids,
+        }
+        if len(compile_target_ids) > 1:
+            target_ids["buildFiles"] = compile_target_ids
+
+        guid_target_ids[target_pif["guid"]] = target_ids
 
     os.makedirs(guid_target_ids_parent, exist_ok = True)
     with open(guid_target_ids_path, "w", encoding = "utf-8") as f:
         json.dump(guid_target_ids, f)
 
     return guid_target_ids
 
+def _platform_from_build_key(key):
+    if key.startswith("BAZEL_TARGET_ID[sdk="):
+        return key[20:-2]
+    return ""
+
+def _platform_from_compile_key(key):
+    if key.startswith("BAZEL_COMPILE_TARGET_ID[sdk="):
+        return key[28:-2]
+    return ""
+
+def _select_target_id(target_ids, platform):
+    key = target_ids["key"]
+
+    platforms = {platform: None}
+
+    # We need to try other similar platforms (i.e. other simulator platforms if
+    # `platform`` is for a simulator). This is to support schemes with targets
+    # of multiple platforms in them. Because `dict` is insertion ordered,
+    # `platform` will be checked first.
+    platforms.update(_similar_platforms(platform))
+
+    for platform in platforms:
+        target_id = target_ids.get(platform)
+        if target_id:
+            if target_id == f"$({key})":
+                return target_ids[""]
+            return target_id
+    return target_ids[""]
+
+def _similar_platforms(platform):
+    if platform == "macosx" or "simulator" in platform:
+        return _SIMULATOR_PLATFORMS
+    return _DEVICE_PLATFORMS
+
 def _main(objroot, base_objroot, scheme_target_id_file, prefix):
     if not os.path.exists(scheme_target_id_file):
         return
 
     with open(scheme_target_id_file, encoding = "utf-8") as f:
         scheme_target_ids = set(f.read().splitlines())
 
-    if not scheme_target_ids:
-        return
-
-    try:
-        build_request_file = _calculate_build_request_file(objroot)
-        guid_target_ids = _calculate_guid_target_ids(base_objroot)
-    except Exception:
-        print(
-            f"""\
+    if prefix == "i":
+        # buildRequest for Index Build includes all targets, so we have to
+        # fallback to the scheme target ids (which are actually set by the
```

Comment:
Nit used as a verb, not a noun.

```suggestion
        # fall back to the scheme target ids (which are actually set by the
```